cmd/fleetagent/main.go

```diff
@@ -56,7 +56,7 @@ func (a *FleetAgent) Run(cmd *cobra.Command, args []string) error {
 		return err
 	}
 	<-cmd.Context().Done()
-	return cmd.Context().Err()
```

Comment:
@moio I considered that. But, what other situation would cause the context to cancel besides a signal? There may be other errors that happen and other signals but I didn't trace any of them ended up in the context error. It appears, unless I missed something, that the context error will only be the cancel we expect.

Returning nil is what the other controllers in Rancher do for the same situation. I moved to the existing pattern.

Reply:
I agree, no other situation in the current code AFAICS. I am fine to keep it simple for now, and complicate later if we have a need.